router/internal/metric/agent.go

```diff
@@ -31,33 +35,65 @@ func createPromExporter() (*prometheus.Exporter, error) {
 	return prometheusExporter, nil
 }
 
-func createHttpExporter(c *Config) (sdkmetric.Exporter, error) {
-	u, err := url.Parse(c.Endpoint)
+func createOTELExporter(log *zap.Logger, exp *OpenTelemetryExporter) (sdkmetric.Exporter, error) {
+	u, err := url.Parse(exp.Endpoint)
 	if err != nil {
-		return nil, fmt.Errorf("invalid OpenTelemetry endpoint: %w", err)
+		return nil, fmt.Errorf("invalid OpenTelemetry endpoint %q: %w", exp.Endpoint, err)
 	}
+	var exporter sdkmetric.Exporter
+	switch exp.Exporter {
+	case otelconfig.ExporterDefault, otelconfig.ExporterOLTPHTTP:
+		opts := []otlpmetrichttp.Option{
+			// Includes host and port
+			otlpmetrichttp.WithEndpoint(u.Host),
+			otlpmetrichttp.WithCompression(otlpmetrichttp.GzipCompression),
+		}
 
-	opts := []otlpmetrichttp.Option{
-		// Includes host and port
-		otlpmetrichttp.WithEndpoint(u.Host),
-		otlpmetrichttp.WithCompression(otlpmetrichttp.GzipCompression),
-	}
+		if u.Scheme != "https" {
+			opts = append(opts, otlpmetrichttp.WithInsecure())
+		}
 
-	if u.Scheme != "https" {
-		opts = append(opts, otlpmetrichttp.WithInsecure())
-	}
+		if len(exp.Headers) > 0 {
+			opts = append(opts, otlpmetrichttp.WithHeaders(exp.Headers))
+		}
+		if len(exp.HTTPPath) > 0 {
+			opts = append(opts, otlpmetrichttp.WithURLPath(exp.HTTPPath))
+		}
+
+		exporter, err = otlpmetrichttp.New(
+			context.Background(),
+			opts...,
+		)
+	case otelconfig.ExporterOLTPGRPC:
```

Comment:
Did you test GRPC integration?

Reply:
I did, this does send the metrics over gRPC:

```yaml
telemetry:
    use_default_exporters: false
    tracing:
        enabled: true
        exporters:
            - endpoint: http://localhost:4317
              exporter: otlpgrpc
```